gui/src-tauri/src/main.rs

```diff
@@ -309,7 +337,38 @@ fn setup_tauri(
 						.emit("server-status", ("other", "receiver cancelled"))
 						.expect("Failed to emit");
 				});
+			};
+
+			let find_server = || -> Option<ServerStartInfo> {
+				use const_format::formatcp;
+
+				let Some(server_jar) = find_server_jar(&cli) else {
+					return None;
+				};
+				log::info!("Server found on path: {}", server_jar.to_str().unwrap());
+
+				let shared_dir = server_jar.parent().unwrap();
+				let Some(java_bin) = find_java_bin(shared_dir) else {
+					show_error(formatcp!(
+						"Couldn't find a compatible Java version, please download Java {} or higher",
+						MINIMUM_JAVA_VERSION
+					));
+					return None;
+				};
+				log::info!("Using Java binary: {:?}", java_bin);
+
+				Some(ServerStartInfo {
+					java_bin,
+					server_jar,
+				})
```

Comment:
I think this is fine as it is? I don't really know Rust well enough, but it would probably be best to keep the order the same as the struct yeah?
```suggestion
				Some(ServerStartInfo {
					server_jar,
					java_bin,
				})
```